shell_integration/MacOSX/NextcloudIntegration/NCDesktopClientSocketKit/LocalSocketClient.m

```diff
@@ -326,24 +326,43 @@ - (void)readFromSocket
 
 - (void)processInBuffer
 {
-    NSLog(@"Processing in buffer. In buffer length %li", [_inBuffer length]);
+    NSLog(@"Processing in buffer. In buffer length %li", _inBuffer.length);
+
     UInt8 separator[] = {0xa}; // Byte value for "\n"
-    while(true) {
-        NSRange firstSeparatorIndex = [_inBuffer rangeOfData:[NSData dataWithBytes:separator length:1] options:0 range:NSMakeRange(0, [_inBuffer length])];
-        
-        if(firstSeparatorIndex.location == NSNotFound) {
-            NSLog(@"No separator found. Stopping.");
-            return; // No separator, nope out
-        } else {
-            unsigned char *buffer = [_inBuffer mutableBytes];
-            buffer[firstSeparatorIndex.location] = 0; // Add NULL terminator, so we can use C string methods
-            
-            NSString *newLine = [NSString stringWithUTF8String:[_inBuffer bytes]];
+    NSData * const separatorData = [NSData dataWithBytes:separator length:1];
+
+    while(_inBuffer.length > 0) {
+        const NSUInteger inBufferLength = _inBuffer.length;
+        const NSRange inBufferLengthRange = NSMakeRange(0, inBufferLength);
+        const NSRange firstSeparatorIndex = [_inBuffer rangeOfData:separatorData
+                                                           options:0
+                                                             range:inBufferLengthRange];
+
+        unsigned char *buffer = _inBuffer.mutableBytes;
+        NSUInteger nullTerminatorIndex = NSUIntegerMax;
 
-            [_inBuffer replaceBytesInRange:NSMakeRange(0, firstSeparatorIndex.location + 1) withBytes:NULL length:0];
-            [_lineProcessor process:newLine];
+        if (firstSeparatorIndex.location == NSNotFound) {
+            NSLog(@"No separator found. Creating new buffer qith space for null terminator.");
+            unsigned char newBuffer[inBufferLength + 1];
+            memcpy(&newBuffer, buffer, inBufferLength);
+            buffer = &newBuffer[0];
```

Comment:
@claucambra Once you exit the if, `unsigned char newBuffer` will get deallocated (as it is on the stack), hence `buffer` will be pointing to an invalid memory location

Reply:
What a thing to miss -- thanks! Fixed now